### PR TITLE
RATIS-1741 Add a removeReporterRegistration method

### PR DESCRIPTION
**Pull request by sumitagrawl**
## What changes were proposed in this pull request?

removeReporterRegistration method is provided

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/RATIS-1741

## How was this patch tested?

Patch is tested adding Unit test with register and remove registration of reporter

### CODE REVIEW

**Comment by sumitagrawl** [comment]
@szetszwo plz review

File: ratis-metrics-api/src/main/java/org/apache/ratis/metrics/MetricRegistries.java [hunk 1]
```diff
@@ -93,10 +93,20 @@ public static MetricRegistries global() {
...
       Consumer<RatisMetricRegistry> stopReporter);
 
+  /**
+   * Remove hook of reporter for the metricRegistry.
+   *
+   * @param reporterRegistration Consumer to create the reporter for the registry.
+   * @param stopReporter Consumer to stop the reporter for the registry.
+   */
+  public abstract void removeReporterRegistration(Consumer<RatisMetricRegistry> reporterRegistration,
+                                               Consumer<RatisMetricRegistry> stopReporter);
```
Comment (szetszwo) [inline ratis-metrics-api/src/main/java/org/apache/ratis/metrics/MetricRegistries.java:108]: Let's add a default implementation for compatibility.
```java
  public void removeReporterRegistration(Consumer<RatisMetricRegistry> reporterRegistration,
      Consumer<RatisMetricRegistry> stopReporter) {
  }
```

**Review by szetszwo — COMMENTED** [review]
@sumitagrawl , thanks a lot for working on this!

We also need to implement it in Dm3MetricRegistriesImpl
```java
+++ b/ratis-metrics-dropwizard3/src/main/java/org/apache/ratis/metrics/dropwizard3/Dm3MetricRegistriesImpl.java
@@ -113,6 +113,13 @@ public class Dm3MetricRegistriesImpl extends MetricRegistries {
     this.stopReporters.add(stopReporter);
   }
 
+  @Override
+  public void removeReporterRegistration(Consumer<RatisMetricRegistry> reporterRegistration,
+      Consumer<RatisMetricRegistry> stopReporter) {
+    this.reporterRegistrations.remove(reporterRegistration);
+    this.stopReporters.remove(stopReporter);
+  }
+
   @Override
   public void enableJmxReporter() {
     addReporterRegistration(
```

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**